### PR TITLE
Add linalg methods needed by FisherLDA and KernelPCA (CPU-only)

### PR DESCRIPTION
**Pull request by micmn**
Needed by #3826 #3842
This includes: add_scalar, center_matrix, eigen_solver, eigen_solver_symmetric, qr_solver, svd, transpose_matrix, triangular_solver.
Some methods (svd, qr, eigen) are not implemented in general fashion but they implement just what is needed by flda, kpca.

~Not merge ready: unit tests missing.~

### CODE REVIEW

**Comment by karlnapf** [comment]
Great!
@OXPHOS please read carefully! :)

**Comment by karlnapf** [comment]
I think this will have to be split into multiple PRs later.
But we can start discussing here

**Comment by vigsterkr** [comment]
@micmn so there are 2 things
a) we need to modularize our linalg header as this is unscalable like this; not only because of compiler running out of memory but because of the fact that the LoC is 1000+ is just insane imo
b) unit tests are missing

**Comment by OXPHOS** [comment]
Good job! I didn't see anything particular requiring changing apart from what wiking mentioned. I guess I'll open a new issue to discuss linalg lib structure / memory usage problem?

File: src/shogun/mathematics/linalg/LinalgBackendEigen.h [hunk 1]
```diff
@@ -397,6 +478,36 @@ class LinalgBackendEigen : public LinalgBackendBase
...
+	{
+		typename SGVector<T>::EigenVectorXtMap a_eig = a;
+		a_eig = a_eig.array() + b;
+	}
+
+	/** Eigen3 matrix add scalar method */
+	template <typename T>
+	void add_scalar_impl(SGMatrix<T>& a, T b) const
+	{
+		typename SGMatrix<T>::EigenMatrixXtMap a_eig = a;
```
Comment (OXPHOS) [inline src/shogun/mathematics/linalg/LinalgBackendEigen.h]: Note: A lot of methods can be merged by simply renaming `SGVector<T>::EigenVectorXtMap` to `SGVector<T>::EigenXtMap`and `SGMatrix<T>::EigenMatrixXtMap` to `SGMatrix<T>::EigenXtMap`, if it won't cause any confusion @karlnapf @vigsterkr

**Comment by vigsterkr** [comment]
yeah is there any reason why we wouldn't do that ASAP?

**Comment by vigsterkr** [comment]
@micmn could you try to rebase on latest develop this as i'm hoping that with the latest appveyor.yaml the out of heap space thing will not occur (fingerscrossed)

**Comment by vigsterkr** [comment]
argh it still fails with #3844 :((( reached out to appveyor, in case they have an idea....

**Pull request merged**